zone/doors.cpp

```diff
@@ -190,6 +190,204 @@ bool Doors::Process()
 	return true;
 }
 
+bool Doors::EnterPendingGuildInstance(Client* sender)
+{
+	if (!sender)
+		return false;
+
+	if (Distance(this->GetPosition(), sender->GetPosition()) > RuleI(Quarm, AcceptableDistanceFromInstanceBook))
+	{
+		sender->Message(Chat::Red, "You are unable to enter an instance because you are too far away from the instance entrance.");
+		return false;
+	}
+
+	uint32 zoneguildid = GUILD_NONE;
+
+	uint32 ourZoneInstanceID = GUILD_NONE;
+	uint8 keepoffkeyring = GetNoKeyring();
+	uint32 keyneeded = GetKeyItem();
+	uint32 zoneid = database.GetZoneID(destination_zone_name);
+	uint32 key = 0;
+	float temp_x = m_destination.x;
+	float temp_y = m_destination.y;
+	if (!DoorKeyCheck(sender, key))
+	{
+		sender->Message(Chat::Red, "You lack a key to enter this zone.");
+		return false;
+	}
+
+	if (sender->IsLockedOutOfInstance(zoneid))
+	{
+
+		// TODO: GetTargetZoneInstanceID / Get raid leader's instance ID if not leader / if leader and not locked out, issue new ID, if not, message
+		Raid* player_raid = sender->GetRaid();
+
+		if (!player_raid)
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid.");
+			return false;
+		}
+
+		if (player_raid->MeetsInstancedRaidRequirements() < RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement))
+		{
+			sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid with %i players at or above level %i present total.",
+				RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
+				RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
+			return false;
+		}
+
+		ourZoneInstanceID = sender->GetTargetZoneInstanceID(zoneid);
+		zoneguildid = ourZoneInstanceID;
+
+		int64 curzoneLockout = zone->GetZoneInstanceLockoutByCharacterAndZone(sender->CharacterID(), zoneid);
+		//TODO: hardcode, retrieve from db later
+		int64 zonelockout = RuleI(Quarm, InstanceMinimumLockoutTime);
+
+		auto cur_time = time(nullptr);
```

Comment:
there's a variety of local variable naming conventions in this function, it might be nice to make them consistent for readability